backend/src/services/comparisons/comparisons.service.ts

```diff
@@ -0,0 +1,194 @@
+        user_id: userId,
+      },
+    });
+
+    await prisma.comparisons_Complectations.create({
+      data: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+        position: 1,
+      },
+    });
+    return comparison;
+  }
+  return activeComparison;
+};
+
+const changeComparisonType = async ({
+  type,
+  userId,
+}: {
+  type: ComparisonType;
+  userId: string;
+}): Promise<Comparison> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  return await prisma.comparison.update({
+    where: { id: comparison.id },
+    data: { type },
+  });
+};
+
+const clearComparison = async (userId: string): Promise<Comparison> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  return await prisma.comparison.update({
+    where: { id: comparison.id },
+    data: { active: false },
+  });
+};
+
+const deleteCarFromComparison = async ({
+  complectationId,
+  userId,
+}: {
+  complectationId: string;
+  userId: string;
+}): Promise<void> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  await prisma.comparisons_Complectations.delete({
+    where: {
+      comparison_id_complectation_id: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+      },
+    },
+  });
+
+  const complectationsAmount = await prisma.comparisons_Complectations.count({
+    where: { comparison_id: comparison.id },
+  });
+
+  if (!complectationsAmount) {
+    await prisma.comparison.delete({
+      where: { id: comparison.id },
+    });
+  }
+};
+
+const getActiveComparisonCars = async (
+  userId: string,
+): Promise<(Complectation | null)[]> => {
+  const activeComparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!activeComparison) {
+    return [];
+  }
+
+  const complectations = await prisma.comparisons_Complectations.findMany({
+    where: { comparison_id: activeComparison.id },
+    select: { complectation_id: true, position: true },
+  });
+
+  const sortedComplectations = complectations.sort((a, b) =>
+    a.position > b.position ? 1 : -1,
+  );
+
+  return Promise.all(
+    sortedComplectations.map(async (car) => {
+      return await prisma.complectation.findUnique({
+        where: { id: car.complectation_id },
+      });
+    }),
+  );
```

Comment:
Oh-oh-oh... There are multiple mistakes here. Can we have a huddle on it?

Reply:
But I'll leave this for the future - https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#in